codegen/templates/tchannel_client.tmpl

```diff
@@ -167,24 +167,49 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 		}
 	}
 
-	client := zanzibar.NewTChannelClientContext(
-		channel,
-		deps.Default.ContextLogger,
-		deps.Default.ContextMetrics,
-		deps.Default.ContextExtractor,
-		&zanzibar.TChannelClientOption{
-			ServiceName:          serviceName,
-			ClientID:             "{{$clientID}}",
-			MethodNames:          methodNames,
-			Timeout:              timeout,
-			TimeoutPerAttempt:    timeoutPerAttempt,
-			RoutingKey:           &routingKey,
-			RuleEngine:           re,
-			HeaderPatterns:       headerPatterns,
-			RequestUUIDHeaderKey: requestUUIDHeaderKey,
-			AltChannelMap:        altChannelMap,
-		},
-	)
+	var client *zanzibar.TChannelClient
+
+	if  deps.Default.Config.ContainsKey("tchannelclients.retryCount.feature.enabled") && deps.Default.Config.MustGetBoolean("tchannelclients.retryCount.feature.enabled") && deps.Default.Config.ContainsKey("clients.{{$clientID}}.retryCount") && int(deps.Default.Config.MustGetInt("clients.{{$clientID}}.retryCount")) > 0{
+		maxAttempts := int(deps.Default.Config.MustGetInt("clients.{{$clientID}}.retryCount"))
+		client = zanzibar.NewTChannelClientContext(
```

Comment:
1) Instead of having this logic in template which will increase code-gen + template fill, you should move this function to tchannel_client.go or someplace and just call it from here.

2) If else blocks here are highly repetitive and I think we should be able to parameterize them